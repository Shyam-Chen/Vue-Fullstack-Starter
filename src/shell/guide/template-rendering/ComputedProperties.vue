<template>
  <v-layout column class="my-1">
    <div class="title mb-2">Computed Properties</div>

    <div>
      <div>Existing quantity: {{ existingQuantity.toLocaleString('en-US') }}</div>
      <div>Required quantity: {{ requiredQuantity.toLocaleString('en-US') }}</div>
      <div>Percentage: {{ percentage }}%</div>
    </div>
  </v-layout>
</template>

<script>
// @flow

type Data = {
  existingQuantity: number,
  requiredQuantity: number,
};

export default {
  data(): Data {
    return {
      existingQuantity: 372900,
      requiredQuantity: 435900,
    };
  },
  computed: {
    percentage(): number {
      return Math.round((this.existingQuantity / this.requiredQuantity) * 100);
    },
  },
};
</script>

<style scoped>
</style>
