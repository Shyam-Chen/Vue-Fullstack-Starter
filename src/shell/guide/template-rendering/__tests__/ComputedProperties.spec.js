import { shallowMount } from '@vue/test-utils';

import ComputedProperties from '../ComputedProperties';

describe('ComputedProperties', () => {
  let [wrapper] = [];

  beforeEach(() => {
    wrapper = shallowMount(ComputedProperties);
  });

  it('should render an initial component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
