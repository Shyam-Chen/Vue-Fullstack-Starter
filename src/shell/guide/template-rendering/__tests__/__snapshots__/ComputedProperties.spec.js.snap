// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComputedProperties should render an initial component 1`] = `
<v-layout-stub tag="div" column="" class="my-1">
  <div class="title mb-2">Computed Properties</div>
  <div>
    <div>Existing quantity: 372,900</div>
    <div>Required quantity: 435,900</div>
    <div>Percentage: 86%</div>
  </div>
</v-layout-stub>
`;
